src/client/src/Stream.c

```diff
@@ -1836,6 +1842,167 @@ STATUS setNalAdaptationFlags(PKinesisVideoStream pKinesisVideoStream, UINT32 nal
     return retStatus;
 }
 
+/**
+ * Puts a metadata into the stream
+ */
+STATUS putEventMetadata(PKinesisVideoStream pKinesisVideoStream, STREAM_EVENT_TYPE event, PStreamEventMetadata pMetadata)
+{
+    ENTERS();
+    STATUS retStatus = STATUS_SUCCESS;
+    PKinesisVideoClient pKinesisVideoClient = NULL;
+    BOOL streamLocked = FALSE, hasMetadata = TRUE, firstEvent = TRUE, creatingNodes = FALSE;
+    UINT32 packagedSize = 0, totalPackagedSize = 0, metadataQueueSize;
+    UINT32 packagedSizes[MAX_FRAGMENT_METADATA_COUNT] = {0};
+    PSerializedMetadata serializedNodes[MAX_FRAGMENT_METADATA_COUNT] = {0};
+    UINT8 neededNodes = 0;
+    PSerializedMetadata pExistingSerializedMetadata;
+    StackQueueIterator iterator;
+    UINT64 data;
+
+    CHK(pKinesisVideoStream != NULL, STATUS_NULL_ARG);
+    pKinesisVideoClient = pKinesisVideoStream->pKinesisVideoClient;
+
+    // Check if the stream has been stopped
+    CHK(!pKinesisVideoStream->streamStopped, STATUS_STREAM_HAS_BEEN_STOPPED);
+
+    // Lock the stream
+    pKinesisVideoClient->clientCallbacks.lockMutexFn(pKinesisVideoClient->clientCallbacks.customData, pKinesisVideoStream->base.lock);
+    streamLocked = TRUE;
+
+    if(pMetadata == NULL) {
+        hasMetadata = FALSE;
+    }
```

Comment:
Is there a reason this needs to be set inside the lock?  Since the value of `hasMetadata` only depends on a local variable which was passed in `pMetadata` why not just set it at the top where it's declared?

```
BOOL streamLocked = FALSE, hasMetadata = pMetadata == NULL ? FALSE : TRUE, firstEvent = TRUE, creatingNodes = FALSE;
```

Reply:
will do